shell/src/main/java/org/apache/accumulo/shell/commands/ActiveCompactionHelper.java

```diff
@@ -0,0 +1,119 @@
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.accumulo.shell.commands;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.Executors;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import org.apache.accumulo.core.client.IteratorSetting;
+import org.apache.accumulo.core.client.TableNotFoundException;
+import org.apache.accumulo.core.client.admin.ActiveCompaction;
+import org.apache.accumulo.core.client.admin.InstanceOperations;
+import org.apache.accumulo.core.util.Duration;
+import org.apache.accumulo.core.util.NamingThreadFactory;
+
+class ActiveCompactionHelper {
+
+  private static String maxDecimal(double count) {
+    if (count < 9.995)
+      return String.format("%.2f", count);
+    if (count < 99.95)
+      return String.format("%.1f", count);
+    return String.format("%.0f", count);
+  }
+
+  private static String shortenCount(long count) {
+    if (count < 1_000)
+      return count + "";
+    if (count < 1_000_000)
+      return maxDecimal(count / 1_000.0) + "K";
+    if (count < 1_000_000_000)
+      return maxDecimal(count / 1_000_000.0) + "M";
+    return maxDecimal(count / 1_000_000_000.0) + "B";
+  }
+
+  private static String formatActiveCompactionLine(String tserver, ActiveCompaction ac)
+      throws TableNotFoundException {
+    String output = ac.getOutputFile();
+    int index = output.indexOf("tables");
+    if (index > 0) {
+      output = output.substring(index + 6);
+    }
+
+    List<String> iterList = new ArrayList<>();
+    Map<String,Map<String,String>> iterOpts = new HashMap<>();
+    for (IteratorSetting is : ac.getIterators()) {
+      iterList.add(is.getName() + "=" + is.getPriority() + "," + is.getIteratorClass());
+      iterOpts.put(is.getName(), is.getOptions());
+    }
+
+    return String.format(
+        "%21s | %9s | %5s | %6s | %5s | %5s | %15s | %-40s | %5s | %35s | %9s | %s", tserver,
+        Duration.format(ac.getAge(), "", "-"), ac.getType(), ac.getReason(),
+        shortenCount(ac.getEntriesRead()), shortenCount(ac.getEntriesWritten()), ac.getTable(),
+        ac.getTablet(), ac.getInputFiles().size(), output, iterList, iterOpts);
+  }
+
+  private static List<String> activeCompactionsForServer(String tserver,
+      InstanceOperations instanceOps) {
+    List<String> compactions = new ArrayList<>();
+    try {
+      List<ActiveCompaction> acl = new ArrayList<>(instanceOps.getActiveCompactions(tserver));
+      acl.sort((o1, o2) -> (int) (o2.getAge() - o1.getAge()));
+      for (ActiveCompaction ac : acl) {
+        compactions.add(formatActiveCompactionLine(tserver, ac));
+      }
+    } catch (Exception e) {
+      compactions.add(tserver + " ERROR " + e.getMessage());
```

Comment:
I like how this is handled.

Reply:
I'm pretty sure you wrote that code. I just moved it. :smiley_cat: